### PR TITLE
Reset cached indices when resetting cache on SSZ read

### PR DESCRIPTION
**Pull request by arnetheduck**
When deserializing into an existing structure, the cache should be
cleared - goes for json also

### CODE REVIEW

**Comment by zah** [comment]
Can we base this fix on the `version-1.1.0` branch, so it can be included in a release earlier?

**Comment by arnetheduck** [comment]
I'll rebase it to the merge base of the two, should be fine to apply to both then

File: beacon_chain/statediff.nim [hunk 1]
```diff
@@ -76,11 +78,13 @@ func replaceOrAddDecodeEth1Votes[T, U](
...
   for item in votes1:
-    votes0.add item
+    if not votes0.add item:
+      raiseAssert "same limit"
 
 func getMutableValidatorStatuses(state: BeaconState):
     List[ValidatorStatus, Limit VALIDATOR_REGISTRY_LIMIT] =
-  result.setLen(state.validators.len)
+  if not result.setLen(state.validators.len):
+    raiseAssert "same limt as validators"
```
Comment (zah) [inline beacon_chain/statediff.nim:87]:
```suggestion
    raiseAssert "same limit as validators"
```

File: beacon_chain/spec/beaconstate.nim [hunk 2]
```diff
@@ -507,7 +513,8 @@ func get_sorted_attesting_indices_list*(
     state: BeaconState, data: AttestationData, bits: CommitteeValidatorsBits,
     cache: var StateCache): List[uint64, Limit MAX_VALIDATORS_PER_COMMITTEE] =
   for index in get_sorted_attesting_indices(state, data, bits, cache):
-    result.add index.uint64
+    if not result.add index.uint64:
+      raiseAssert "same length as bits"
```
Comment (zah) [inline beacon_chain/spec/beaconstate.nim]:
```suggestion
      raiseAssert "The `result` list has the same max size as the sorted `bits` input"
```

**Pull request merged**